src/test/java/cat/udl/eps/softarch/demo/steps/RetrieveRequestStepDefs.java

```diff
@@ -117,4 +125,17 @@ public void iCanTSeeAnyRequest() throws Exception {
     public void iMNotAllowedToSeeAnyRequest() throws Exception {
         stepDefs.result.andExpect(jsonPath("$").doesNotExist());
     }
+
+    @When("I retrieve a request with name {string}, price {int}, description {string} by {string}")
```

Comment:
For the next sprint, make sure that all "@When" interact with the API, they should test that it provides the expected behaviour. Thus, for the "@When" steps, avoid using the repositories to perform the changes directly.